File not found on make install-python-ci-dependencies-uv step in Developer Guide

## Expected Behavior 
Follow the developer guide to step 9 in the Feast Python SDK / CLI section. Then do step 9, which is to execute the command:
```make install-python-ci-dependencies-uv```

I expect this to run successfully and install dependencies.

## Current Behavior
I run the command from the feast root directory, and I get the error message:
```uv pip sync --system sdk/python/requirements/py-ci-requirements.txt
error: File not found: `sdk/python/requirements/py-ci-requirements.txt`
make: *** [install-python-ci-dependencies-uv] Error 2
```

On inspection, the file `sdk/python/requirements/py-ci-requirements.txt` doesn't exist, but similar files do:
```ls sdk/python/requirements/
# output
py3.10-ci-requirements.txt	py3.11-ci-requirements.txt	py3.9-ci-requirements.txt
py3.10-requirements.txt		py3.11-requirements.txt		py3.9-requirements.txt
```

My guess is, the requirement files were updated, but the make command was not.

## Steps to reproduce
Verifying this should be as easy as running the command `make install-python-ci-dependencies-uv` from the feast root directory.

To reproduce exactly, follow the [Developer Guide](https://docs.feast.dev/project/development-guide#feast-python-sdk-cli) steps to step 9.

### Specifications

- Version: 0.40.1
- Platform: Mac
- Subsystem: M2

## Possible Solution
A couple of suggestions
1. Update the make command to point to an existing requirements file. We would have to pick which one, or have logic behind make determine the python version that is best--perhaps with input from the user.
2. Update the developer guide to run `pip install -r `sdk/python/requirements/py3.9-ci-requirements.txt`, with instruction to substitute the version if needed. (Possibly `uv pip` instead of just `pip`).

--- a/Makefile
+++ b/Makefile
@@ -21,6 +21,7 @@ ifeq ($(shell uname -s), Darwin)
 	OS = osx
 endif
 TRINO_VERSION ?= 376
+PYTHON_VERSION = ${shell python --version | grep -Eo '[0-9]\.[0-9]+'}
 
 # General
 
@@ -37,22 +38,22 @@ build: protos build-java build-docker
 # Python SDK
 
 install-python-ci-dependencies:
-	python -m piptools sync sdk/python/requirements/py$(PYTHON)-ci-requirements.txt
+	python -m piptools sync sdk/python/requirements/py$(PYTHON_VERSION)-ci-requirements.txt
 	pip install --no-deps -e .
 	python setup.py build_python_protos --inplace
 
 install-python-ci-dependencies-uv:
-	uv pip sync --system sdk/python/requirements/py$(PYTHON)-ci-requirements.txt
+	uv pip sync --system sdk/python/requirements/py$(PYTHON_VERSION)-ci-requirements.txt
 	uv pip install --system --no-deps -e .
 	python setup.py build_python_protos --inplace
 
 install-python-ci-dependencies-uv-venv:
-	uv pip sync sdk/python/requirements/py$(PYTHON)-ci-requirements.txt
+	uv pip sync sdk/python/requirements/py$(PYTHON_VERSION)-ci-requirements.txt
 	uv pip install --no-deps -e .
 	python setup.py build_python_protos --inplace
 
 lock-python-ci-dependencies:
-	uv pip compile --system --no-strip-extras setup.py --extra ci --output-file sdk/python/requirements/py$(PYTHON)-ci-requirements.txt
+	uv pip compile --system --no-strip-extras setup.py --extra ci --output-file sdk/python/requirements/py$(PYTHON_VERSION)-ci-requirements.txt
 
 package-protos:
 	cp -r ${ROOT_DIR}/protos ${ROOT_DIR}/sdk/python/feast/protos
@@ -61,11 +62,11 @@ compile-protos-python:
 	python setup.py build_python_protos --inplace
 
 install-python:
-	python -m piptools sync sdk/python/requirements/py$(PYTHON)-requirements.txt
+	python -m piptools sync sdk/python/requirements/py$(PYTHON_VERSION)-requirements.txt
 	python setup.py develop
 
 lock-python-dependencies:
-	uv pip compile --system --no-strip-extras setup.py --output-file sdk/python/requirements/py$(PYTHON)-requirements.txt
+	uv pip compile --system --no-strip-extras setup.py --output-file sdk/python/requirements/py$(PYTHON_VERSION)-requirements.txt
 
 lock-python-dependencies-all:
 	pixi run --environment py39 --manifest-path infra/scripts/pixi/pixi.toml "uv pip compile --system --no-strip-extras setup.py --output-file sdk/python/requirements/py3.9-requirements.txt"

--- a/docs/project/development-guide.md
+++ b/docs/project/development-guide.md
@@ -132,8 +132,7 @@ Setting up your development environment for Feast Python SDK / CLI:
   source venv/bin/activate
   ```
 4. (M1 Mac only): Follow the [dev guide](https://github.com/feast-dev/feast/issues/2105)
-5. Install uv
-It is recommended to use uv for managing python dependencies.
+5. Install uv. It is recommended to use uv for managing python dependencies.
 ```sh
 curl -LsSf https://astral.sh/uv/install.sh | sh
 ```
@@ -145,21 +144,26 @@ pip install uv
 ```
 make build-ui
 ```
-7. (Optional) install pixi
-pixi is necessary to run step 8 for all python versions at once.
+7. (Optional) install pixi. pixi is necessary to run step 8 for all python versions at once.
 ```sh
 curl -fsSL https://pixi.sh/install.sh | bash
 ```
-8. (Optional): Recompile python lock files
-If you make changes to requirements or simply want to update python lock files to reflect latest versioons.
+8. (Optional): Recompile python lock files. Only when you make changes to requirements or simply want to update python lock files to reflect latest versioons.
 ```sh
 make lock-python-dependencies-all
 ``` 
-9. Install development dependencies for Feast Python SDK / CLI
-This will install package versions from the lock file, install editable version of feast and compile protobufs.
+9. Install development dependencies for Feast Python SDK / CLI. This will install package versions from the lock file, install editable version of feast and compile protobufs.
+
+If running inside a virtual environment:
+```sh
+make install-python-ci-dependencies-uv-venv
+```
+
+Otherwise:
 ```sh
 make install-python-ci-dependencies-uv
 ```
+
 10. Spin up Docker Image
 ```sh
 docker build -t docker-whale -f ./sdk/python/feast/infra/feature_servers/multicloud/Dockerfile .

